### PR TITLE
When we have an interface with static IP configuration and no cable p…

### PR DESCRIPTION
**Pull request by gkodali-zededa**
…lugged into it, dhcpcd would exit. As a result we enter an infinite loop in NIM main thread. Similar to how we retry only thrice in case of DHCP configuration, we should the same for static IP configuration also

Signed-off-by: GopiKrishna Kodali <gkodali@zededa.com>

### CODE REVIEW

File: pkg/pillar/devicenetwork/dhcpcd.go [hunk 1]
```diff
@@ -165,11 +165,22 @@ func doDhcpClientActivate(nuc types.NetworkPortConfig) bool {
...
+		failed := false
 		for !dhcpcdExists(nuc.IfName) {
 			log.Warnf("dhcpcd %s not yet running", nuc.IfName)
+			waitCount++
+			if waitCount >= 3 {
+				log.Errorf("dhcpcd %s not yet running", nuc.IfName)
+				failed = true
+				break
+			}
 			time.Sleep(10 * time.Second)
```
Comment (eriknordmark) [inline pkg/pillar/devicenetwork/dhcpcd.go]: Do you want to sleep for less so that the loop will complete in less than 30 seconds? We try to make sure all operations finish in 30 seconds or so. sleep(2) might be sufficient??

File: pkg/rsyslog/rsyslog.conf [hunk 2]
```diff
@@ -55,12 +55,14 @@ action(type="mmjsonparse" cookie="")
```
Comment (rvs) [inline pkg/rsyslog/rsyslog.conf]: This is not really needed for this PR, right? I mean it could be a fine change in its own PR but not required for the NIM part, correct @gkodali-zededa ?

Comment (gkodali-zededa) [reply pkg/rsyslog/rsyslog.conf]: @rvs Yes, this is not required for this PR but the I saw this also at the same time. Should I remove this change and raise a new PR?

File: pkg/pillar/devicenetwork/dhcpcd.go (hunk 1 above)
Comment (gkodali-zededa) [reply pkg/pillar/devicenetwork/dhcpcd.go]: @eriknordmark 30 seconds is what DH_CLIENT case also uses. My thinking was that there should have been a reason why 30 seconds is being used for DH_CLIENT case, that is the reason I opted to use the same wait time for DH_STATIC also. May be I can reduce the sleep to 15 seconds ? 5 seconds in each iteration?

Comment (eriknordmark) [reply pkg/pillar/devicenetwork/dhcpcd.go]: Aren't you also waiting in the code which calls this dhcp routine? If you do, then what is the total time you can wait?
Note that what you have is better than now, since currently it waits forever.

Comment (gkodali-zededa) [reply pkg/pillar/devicenetwork/dhcpcd.go]: @eriknordmark While testing new DPC we have the PendTimer (default 30 secs) that fires VerifyPending. But as part of this we check if there are any usable IP addresses and not for existence of dhcpcd for the port of interest. Adding both I think in the worst case we wait for 60 seconds in a DPC with one port. But the PendTimer (30 secs) in non-blocking. The waiting that we do as part do doDhcpClientActivate (30 secs) is blocking though.

File: pkg/rsyslog/rsyslog.conf (hunk 2 above)
Comment (gkodali-zededa) [reply pkg/rsyslog/rsyslog.conf]: @rvs I backed out the rsyslog related commit. Will send as part of another PR.
